### PR TITLE
1231 test case all the teasers in the app

### LINKED ISSUE #1231: test case: all the teasers in the app.

### PR DESCRIPTION
**Pull request by Ekaanth**

### CODE REVIEW

File: src/ui/components/BountySummaryTeaser.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+describe('BountySummaryTeaser', () => {
+  it('should render the loading view while data is fetching', () => {
+    const {getByTestId} = render(<BountySummaryTeaser onPress={onPressEvent.navigate} />);
+    expect(getByTestId('loading_box')).toBeTruthy();
+  });
+
+  it('should render the bountySummaryTeaser component with initial data', async () => {
+    const {findByText} = render(<BountySummaryTeaser onPress={onPressEvent.navigate} />);
+    await findByText('Bounties');
+    await findByText('Active');
```
Comment (silva-fj) [inline src/ui/components/BountySummaryTeaser.test.tsx:19]: Let's also make assertions on the values (the numbers), not just the labels in the teaser

File: src/ui/components/BountySummaryTeaser.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    await findByText('Past');
+    await findByText('Active total');
+    await findByText('Funding period (17 days)');
+  });
+
+  it('should navigate to bounty summary page on click of the teaser', async () => {
```
Comment (silva-fj) [inline src/ui/components/BountySummaryTeaser.test.tsx]: This test is a bit misleading, here we are not actually testing navigation, to test navigation you would have to pass the `NavigationContainer` and the routes setup with the screens and everything, like in [this example](https://callstack.github.io/react-native-testing-library/docs/react-navigation#example-tests).

The `BountySummaryTeaser` component does not know anything about navigation, it just takes an `onPress` that is called when the container is pressed. The test is fine, we just need to change the description

File: src/ui/components/CouncilSummaryTeaser.test.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+describe('CouncilSummaryTeaser', () => {
+  it('should render the loading box component while data is fetching', () => {
+    const {getByTestId} = render(<CouncilSummaryTeaser onPress={onPressEvent.navigate} />);
+    expect(getByTestId('loading_box')).toBeTruthy();
+  });
+
+  it('should render the CouncilSummaryTeaser component when data is fetched', async () => {
+    const {findByText} = render(<CouncilSummaryTeaser onPress={onPressEvent.navigate} />);
+    await findByText('Council');
+    await findByText('Seats');
```
Comment (silva-fj) [inline src/ui/components/CouncilSummaryTeaser.test.tsx:18]: Let's add the values in these assertions as well, we know them from the mocked data

File: src/ui/components/CouncilSummaryTeaser.test.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    await findByText('Runners up');
+    await findByText('Candidates');
+    await findByText('Term Progress (1 day)');
+    await findByText('Prime Voter');
+  });
+
+  it('should navigate to council summary page on click of the teaser', async () => {
```
Comment (silva-fj) [inline src/ui/components/CouncilSummaryTeaser.test.tsx]: Same case as before. 

also Nitpicking, click is for web, on a mobile app you tap or press.

File: src/ui/components/CrowdloanSummaryTeaser.test.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('CrowdloanSummaryTeaser', () => {
+  it('should render the loading box while data is fetching', () => {
+    const {getByTestId} = render(<CrowdloanSummaryTeaser />);
+    expect(getByTestId('loading_box')).toBeTruthy();
+  });
+
+  it('should render the CrowdloanSummaryTeaser component when data is fetched', async () => {
+    const {findByText} = render(<CrowdloanSummaryTeaser />);
+    await findByText('Active Raised / Cap');
```
Comment (silva-fj) [inline src/ui/components/CrowdloanSummaryTeaser.test.tsx:13]: Same here, let's make assertions on the mocked data

File: src/ui/components/DemocracySummaryTeaser.test.tsx [hunk 6]
```diff
@@ -1,21 +1,32 @@
...
-  expect(getByTestId('loading_box')).toBeTruthy();
-});
+describe('DemocracySummaryTeaser', () => {
+  it('should render the loading view while data is fetching', () => {
+    const {getByTestId} = render(<DemocracySummaryTeaser onPress={onPressEvent.navigate} />);
+    expect(getByTestId('loading_box')).toBeTruthy();
+  });
+
+  test('should render the component data after the loading view has finished', async () => {
+    const {findByText, findAllByText} = render(<DemocracySummaryTeaser onPress={onPressEvent.navigate} />);
```
Comment (silva-fj) [inline src/ui/components/DemocracySummaryTeaser.test.tsx:16]: Same here

File: src/ui/components/DemocracySummaryTeaser.test.tsx [hunk 7]
```diff
@@ -1,21 +1,32 @@
...
 
-test('should render the component data after the loading view has finished', async () => {
-  const {getByText, getAllByText} = render(<DemocracySummaryTeaser onPress={onPressEvent} />);
-  await waitFor(() => {
-    expect(getByText('Democracy')).toBeDefined();
-    expect(getByText('Referenda')).toBeDefined();
-    expect(getAllByText('Total').length).toBe(2);
-    expect(getByText('Launch period')).toBeDefined();
-    expect(getByText('26% 20 days 11 hrs')).toBeDefined();
+  it('should navigate to council summary page on click of the teaser', async () => {
```
Comment (silva-fj) [inline src/ui/components/DemocracySummaryTeaser.test.tsx]: Same here

File: src/ui/components/TreasurySummaryTeaser.test.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import {TreasurySummaryTeaser} from './TreasurySummaryTeaser';
+
+const onPressEvent = jest.fn();
+
+test('render the loading view while data is fetching', () => {
+  const {getByTestId} = render(<TreasurySummaryTeaser onPress={onPressEvent} />);
+  expect(getByTestId('loading_box')).toBeTruthy();
+});
+
+test('should render the teaser component when data is fetched', async () => {
```
Comment (silva-fj) [inline src/ui/components/TreasurySummaryTeaser.test.tsx]: same here

File: src/ui/screens/DashboardScreen.test.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+jest.mock('src/hooks/useParachainAppEnabled', () => ({
+  useParachainAppEnabled: () => {
+    return {parachainAppEnabled: false};
+  },
+}));
+
+describe('DashboardScreen', () => {
+  it('should render all the loading view while data is fetching', () => {
+    const {getAllByTestId} = render(<DashboardScreen navigation={navigation} route={route} />);
+    expect(getAllByTestId('loading_box').length).toBe(5);
```
Comment (silva-fj) [inline src/ui/screens/DashboardScreen.test.tsx:20]: As we discussed on slack, the loading_box is an implementation detail of the *Teaser components. Changing one of the teasers implementation should not break this test

File: src/ui/screens/DashboardScreen.test.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  test('should render various teaser component when data is fetched', async () => {
```
Comment (silva-fj) [inline src/ui/screens/DashboardScreen.test.tsx]: Let's split this into individual tests or use a [data set ](https://jestjs.io/blog/2018/05/29/jest-23-blazing-fast-delightful-testing#jest-each). 

```
const sections ['Upcoming events', ...];

test.each(sections)("should display the section %s", (section) => {...})
```

File: src/ui/components/TreasurySummaryTeaser.test.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import {TreasurySummaryTeaser} from './TreasurySummaryTeaser';
+
+const onPressEvent = jest.fn();
+
+test('render the loading view while data is fetching', () => {
+  const {getByTestId} = render(<TreasurySummaryTeaser onPress={onPressEvent} />);
+  expect(getByTestId('loading_box')).toBeTruthy();
+});
+
+test('test onPress prop functionality', async () => {
```
Comment (silva-fj) [inline src/ui/components/TreasurySummaryTeaser.test.tsx]: What are you testing here for onPress?

File: src/ui/screens/DashboardScreen.test.tsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    const {findByText} = render(<DashboardScreen navigation={navigation} route={route} />);
+    const events = await findByText('Upcoming events');
+    fireEvent.press(events);
+    expect(navigationSpy).toBeCalledWith('Events');
+  });
+
+  const sections = ['Democracy', 'Council', 'Treasury', 'Bounties'];
+  it.each(sections)('should test onPress prop functionality of %s', async (expected) => {
+    const {findByText} = render(<DashboardScreen navigation={navigation} route={route} />);
+    const expectedEvent = await findByText(expected);
```
Comment (silva-fj) [inline src/ui/screens/DashboardScreen.test.tsx]: NIT: This is not an event, you can call it `teaser` or `section`

**Review by ishrak-imam — APPROVED** [review]

**Pull request merged**